tests/local/power_bi/test_power_bi.py

```diff
@@ -23,103 +23,1089 @@ def test_get_workspace_success(self, mock_get):
         mock_get.return_value = mock_response
 
         sut = PowerBi(PowerBiClient(), workspace_name="Finance")
-        sut.powerbi_url = "test"
+        sut.powerbi_url = "test/"
 
         # Act
-        result = sut._get_workspace()
+        result = sut._verify_workspace()
 
         # Assert
         self.assertTrue(result)
         self.assertEqual("614850c2-3a5c-4d2d-bcaa-d3f20f32a2e0", sut.workspace_id)
 
     @patch("requests.get")
-    def test_get_workspace_failure(self, mock_get):
+    def test_verify_workspace_failure(self, mock_get):
+        # Arrange
+        mock_response = Mock()
+        mock_response.status_code = 401
+        mock_response.text = "error"
+        mock_get.return_value = mock_response
+
+        sut = PowerBi(PowerBiClient(), workspace_name="Finance")
+        sut.powerbi_url = "test/"
+
+        # Act
+        with self.assertRaises(SpetlrException) as context:
```

Comment:
Just can consider create a subclass to spetlrexception with a more descriptive class name.

MySpecificError(SpetlrException).

Then you can also assert the specific exeption!

But it is just niceness, not neednes

Reply:
Good idea. This is not a breaking change if the new exception inherits from SpetrlException. I will do it.